common/scala/src/main/scala/whisk/connector/kafka/KafkaMessagingProvider.scala

```diff
@@ -84,4 +102,14 @@ object KafkaConfiguration {
   def configMapToKafkaConfig(configMap: Map[String, String]) = configMap.map {
     case (key, value) => configToKafkaKey(key) -> value
   }
+
+  def useSSL(props: Properties, httpsConfig: KafkaSslConfig) = {
+    props.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, "SSL")
+    props.put(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, httpsConfig.truststorePath)
+    props.put(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, httpsConfig.truststorePassword)
+  }
+  def authenticateUser(props: Properties, httpsConfig: KafkaSslConfig) = {
+    props.put(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, httpsConfig.keystorePath)
+    props.put(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, httpsConfig.keystorePassword)
+  }
```

Comment:
I lately did an effort to generalize all the kafka settings for the producer into `whisk.kafka.producer`. Could we introduce a `whisk.kafka.clients` or `whisk.kafka.common` which includes settings needed for both the producer and consumer?

I modeled it in a way so that we can just throw in the keys from the Kafka config itself and override anything we want from application.conf without needing code changes. Do you think that's feasible here?

Reply:
Yeap, sounds feasible. I could have probably seen it somewhere in the code, will update it